Framework/Geometry/src/Instrument/ComponentInfo.cpp

```diff
@@ -107,8 +107,33 @@ ComponentInfo::componentsInSubtree(size_t componentIndex) const {
   return m_componentInfo->componentsInSubtree(componentIndex);
 }
 
+const std::vector<size_t> &
+ComponentInfo::children(size_t componentIndex) const {
+  return m_componentInfo->children(componentIndex);
+}
+
 size_t ComponentInfo::size() const { return m_componentInfo->size(); }
 
+ComponentInfo::StructuredPanel
+ComponentInfo::structuredPanel(const size_t componentIndex) const {
+  StructuredPanel corners;
```

Comment:
This is the same calculation performed for Bounding Boxes for rectangular banks within this class. Can this be re-factored to make the implementation common somehow?